File: JS/Day_1/reversestring.js
// using a inbuilt function reversestring.js

// function reversestring(str){
// return str.split("").reverse().join("")
// }
// console.log(reversestring("hello"));

function  reverstring(str){
    let reversed = "";
    for(let i=str.length-1;i>=0;i--){
        reversed += str[i];
    }
 return reversed;
}
console.log(reverstring("hello"));

File: Pattern & Recursion/Recursion/reversestring.cpp
#include<iostream>
using  namespace std;
void reversestring(string str, int index){
    if(index < 0 )
    return ;
    cout << str[index];
    reversestring(str, index-1);
}

int main(){
    string str = "Hello";
    reversestring(str ,str.length() -1 );
    return 0;
}